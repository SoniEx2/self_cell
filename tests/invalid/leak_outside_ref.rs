use self_cell::self_cell;

struct Dependent<'a> {
    good: &'a String,
    bad: &'a String,
}

self_cell!(
    struct NoCov {
        #[from_fn]
        owner: String,

        #[covariant]
        dependent: Dependent,
    }
);

fn main() {
    let outside_string = String::from("outside string");

    let _cell = NoCov::from_fn("hi this is no good".into(), |owner| Dependent {
        good: owner,
        bad: &outside_string,
    });
}
