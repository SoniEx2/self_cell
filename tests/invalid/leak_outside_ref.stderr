error[E0597]: `outside_string` does not live long enough
  --> $DIR/leak_outside_ref.rs:23:15
   |
21 |       let _cell = NoCov::from_fn("hi this is no good".into(), |owner| Dependent {
   |  _____________________________________________________________-------_-
   | |                                                             |
   | |                                                             value captured here
22 | |         good: owner,
23 | |         bad: &outside_string,
   | |               ^^^^^^^^^^^^^^ borrowed value does not live long enough
24 | |     });
   | |_____- returning this value requires that `outside_string` is borrowed for `'static`
25 |   }
   |   - `outside_string` dropped here while still borrowed
